_ml-commons-plugin/gpu-acceleration.md

```diff
@@ -0,0 +1,224 @@
+# max virtual memory areas vm.max_map_count to 262144
+sudo sysctl -w vm.max_map_count=262144
+```
+
+#### Amazon Linux2
+
+```
+# Configure Linux for Neuron repository updates
+sudo tee /etc/yum.repos.d/neuron.repo > /dev/null <<EOF
+[neuron]
+name=Neuron YUM Repository
+baseurl=https://yum.repos.neuron.amazonaws.com
+enabled=1
+metadata_expire=0
+EOF
+sudo rpm --import https://yum.repos.neuron.amazonaws.com/GPG-PUB-KEY-AMAZON-AWS-NEURON.PUB
+# Update OS packages
+sudo yum update -y
+################################################################################################################
+# To install or update to Neuron versions 1.19.1 and newer from previous releases:
+# - DO NOT skip 'aws-neuron-dkms' install or upgrade step, you MUST install or upgrade to latest Neuron driver
+################################################################################################################
+# Install OS headers
+sudo yum install kernel-devel-$(uname -r) kernel-headers-$(uname -r) -y
+# Install Neuron Driver
+####################################################################################
+# Warning: If Linux kernel is updated as a result of OS package update
+#          Neuron driver (aws-neuron-dkms) should be re-installed after reboot
+####################################################################################
+sudo yum install aws-neuronx-dkms -y
+# Install Neuron Tools
+sudo yum install aws-neuronx-tools -y
+
+# Install Python venv and activate Python virtual environment to install    
+# Neuron pip packages.
+cd ~
+sudo yum install -y python3.7-venv gcc-c++
+python3.7 -m venv pytorch_venv
+source pytorch_venv/bin/activate
+pip install -U pip
+
+# Set Pip repository  to point to the Neuron repository
+pip config set global.extra-index-url https://pip.repos.neuron.amazonaws.com
+
+#Install Neuron PyTorch
+pip install torch-neuron torchvision
+# If need to trace neuron model, install torch neuron with this command
+# pip install torch-neuron neuron-cc[tensorflow] "protobuf<4" torchvision
+
+# If need to trace neuron model, install transformers for tracing Huggingface model.
+# pip install transformers
+
+# Copy torch neuron lib to OpenSearch
+PYTORCH_NEURON_LIB_PATH=~/pytorch_venv/lib/python3.7/site-packages/torch_neuron/lib/
+mkdir -p $OS_HOME/lib/torch_neuron; cp -r $PYTORCH_NEURON_LIB_PATH/ $OS_HOME/lib/torch_neuron
+export PYTORCH_EXTRA_LIBRARY_PATH=$OS_HOME/lib/torch_neuron/lib/libtorchneuron.so
+echo "export PYTORCH_EXTRA_LIBRARY_PATH=$OS_HOME/lib/torch_neuron/lib/libtorchneuron.so" | tee -a ~/.bash_profile
+# Increase JVm stack size to >=2MB
+echo "-Xss2m" | tee -a $OS_HOME/config/jvm.options
+# Increase max file descriptors to 65535
+echo "$(whoami) - nofile 65535" | sudo tee -a /etc/security/limits.conf
+# max virtual memory areas vm.max_map_count to 262144
+sudo sysctl -w vm.max_map_count=262144
+```
+
+## Install GPU accelerator drivers manually
```

Comment:
The doc structure seems confusing. This part is only for AWS Inferentia.

How about organize by different GPU devices like this doc https://github.com/opensearch-project/ml-commons/blob/2.x/docs/model_serving_framework/GPU_support.md,

Reply:
I'll add note on the header here.

Reply:
Is this copied from AWS documentation, or are there steps specific to OpenSearch? If these steps are purely specific to compiling models on the Neuron SDK, I recommend linking to the AWS documentation. It will avoid the problem of these steps becoming outdated.

Reply:
Decided to keep the step since the Neuron docs don't include on how install the Neuron SDK by itself.